src/postgres-util/src/lib.rs

```diff
@@ -234,16 +246,19 @@ pub async fn publication_info(
                 let scalar_type = match pg_type {
                     PgType::NUMERIC | PgType::NUMERIC_ARRAY => {
                         let modifier: i32 = row.get("modifier");
-                        // https://github.com/postgres/postgres/blob/REL_13_3/src/backend/utils/adt/numeric.c#L978-L983
-                        let tmp_mod = modifier - 4;
-                        let scale = (tmp_mod & 0xffff) as u16;
-                        let precision = ((tmp_mod >> 16) & 0xffff) as u16;
+                        // https://github.com/postgres/postgres/blob/REL_13_3/src/backend/utils/adt/numeric.c#L967-L983
+                        let typ_mod = if modifier < 4 {
+                            None
+                        } else {
+                            let tmp_mod = modifier - 4;
```

Comment:
IMO it's worth us adding a `const PG_HDRSZ: i32 = 4` (or `PG_HEADER_SIZE`) to mirror pg's `VARHDRSZ`.

Reply:
good idea! done